core/actionProxy/actionproxy.py

```diff
@@ -46,59 +47,50 @@ def __init__(self, source = None, binary = None):
     # @return True iff binary exists and is executable
     def init(self, message):
         def prep():
-            if 'code' in message:
+            if 'code' in message and message['code'] is not None:
                 binary = message['binary'] if 'binary' in message else False
                 if not binary:
-                    with codecs.open(self.source, 'w', 'utf-8') as fp:
```

Comment:
Rewrite as you see fit - code improvements and best practices are always appreciated.